### DISCUSSION TITLE
Deployment Telegram Bot

### DISCUSSION THREAD

**Discussion by ChristoNindyo** [teamalgoritma/community#202] [Q&A] [answered]
Saya ingin menanyakan mengenai Capstone Telebot. Saya sudah berhasil menjalankan Telegram Bot dengan Jupyter Notebook, bisa digunakan dengan baik di Telegram. Secara coding sudah tepat. Lalu  saya pindahkan ke python file (.py). Push ke github+heroku. Namun setelah di-push, bot berjalan kurang baik. Kadang bot salah jawab. Bisa minta tolong alasannya apa ya ? Bagaimana cara memperbaikinya ?

Sebelumnya, saya ucapkan terima kasih.

**Comment by yosiaazarya** [comment, marked as answer]
Hi Pak Christo! Hal itu terjadi karena prinsip kerja dari bot kita yang berubah saat kita menjalankan di notebook dan saat kita menjalankan di server. Saat di local kita melakukan polling (bot aktif menembak ke server terus menerus meski tidak ada request), saat di server kita melakukan bantuan webhook dimana bot akan melakukan tembakan ke server untuk service saat ada response berupa perintah di bot kita. Hal ini merupakan kebutuhan kita mengingat kita menggunakan server gratis yang memiliki batas penggunaan dari segi request dan juga RAM per bulannya. Untuk improvement lain agar bot tidak terkesan lag sedang kami usahakan, namun belum menemukan titik terang. Terima kasih banyak pak.

**Reply by ChristoNindyo** [reply]
oke, Mas Yosia. Jadi karena server dan ram ya kalau begitu, Terima kasih atas penjelasannya.
